File: medium/7. Array Of Products/array_of_products.cpp
#include <vector>

std::vector<int> arrayOfProducts(std::vector<int> array) {
  std::vector<int> result(array.size());
  for (int i = 0; i < array.size(); i++) {
    int product = 1;
    for (int j = 0; j < array.size(); j++) {
      if (i != j) {
        product *= array[j];
      }
    }
    result[i] = product;
  }
  return result;
}

File: dsa/medium/7. Array Of Products/array_of_products.go
package main

func ArrayOfProducts(array []int) []int {
	out := make([]int, len(array))
	for i := range array {
		product := 1
		for j := range array {
			if i != j {
				product *= array[j]
			}
		}
		out[i] = product
	}
	return out
}
